libcontainer/process_linux.go

```diff
@@ -395,20 +395,6 @@ func (p *initProcess) start() error {
 					return newSystemErrorWithCause(err, "setting Intel RDT config for procHooks process")
 				}
 			}
-			if p.config.Config.Hooks != nil && len(p.config.Config.Hooks.Prestart) > 0 {
```

Comment:
Here again, this removal obsoletes some of the preceding comments.  And maybe `procHooks` as a whole can go away now?

Reply:
Reading the PR again, the changes have stopped making sense. I think I'm going to have to properly carry this PR...

Reply:
> I think I'm going to have to properly carry this PR...

Any chance we can just land #1741?  I think we all agree on that part of this anyway.

Reply:
I mean, we can land that too (this PR includes it as the first commit). But my reason for wanting this in 1.0 is because currently `runc` doesn't correctly implement hooks -- and that's a bit of an issue if we want to be at least mostly spec compliant.

Reply:
> I mean, we can land that too (this PR includes it as the first commit). But my reason for wanting this in 1.0 is because currently runc doesn't correctly implement hooks -- and that's a bit of an issue if we want to be at least mostly spec compliant.

Right.  I just don't think we have to jump straight to compliant in a single PR.  #1741 moves us a tiny bit closer, and doesn't seem to break anything, so lets land it.  Then you or @dongsupark can pick up the timing fix from this PR and get that landed after you've worked it out.

Reply:
My main reasoning is that time-to-merge for time-critical PRs is usually the main blocker due to timezones. The plan for runc 1.0 is that I send out the release notification this week, and I am at KiwiCon from Thursday on.

But if you reopen the PR and rebase it now I will LGTM it.

Reply:
> But if you reopen the PR and rebase it now I will LGTM it.

I cannot re-open on my own (I need someone with write access to the repo to do that).  Once you re-open, I'll rebase (does it even need a rebase?  When I rebase locally I see no conflicts).

Reply:
Either way I'm fine.

If https://github.com/opencontainers/runc/pull/1741 gets first merged, then I can rebase this PR #1811 on top of that. Also I can fix other issues with obsolete comments.

Or if you want to properly carry this PR with the other one altogether, then you can do so. Go ahead. :-)